docs/features/records.md

```diff
@@ -108,35 +174,44 @@ The programmer may provide an explicit constructor with the same signature as th
+- [ ] **Open issue**: In a record struct, it is an error for there to be no record parameters?
 
-	* Otherwise a base constructor is invoked with no arguments.
+#### Primary constructor body
 
-* executes the body of an explicitly-declared primary constructor, if any.
+```antlr
+primary_constructor_body
+	: attributes? constructor_modifiers? identifier block
+	;
+```
 
-- [ ] **Open issue**: Is this the syntax we want for adding code to the primary constructor?
-- [ ] **Open issue**: Should it be possible to change the access modifier on the primary constructor?
+A *primary_constructor_body* may only be used within a record type declaration. The *identifier* of a *primary_constructor_body* shall name the record type in which it is declared.
 
-#### Properties
+The *primary_constructor_body* does not declare a member on its own, but is a way for the programmer to provide attributes for, and specify the access of, a record type's primary constructor. It also enables the programmer to provide additional code that will be executed when an instance of the record type is constructed.
 
-For each value parameter of a record type declaration there is a corresponding `public` property member whose name and type are taken from the value parameter declaration. If this public property is explicitly declared in the body of the type declaration, it must have the `public` access modifier and a public `get` accessor.
+- [ ] **Open issue**: We should note that a struct default constructor bypasses this.
+- [ ] **Open issue**: We should specify the execution order of initialization.
+- [ ] **Open issue**: Should we allow something like a *primary_constructor_body* (presumably without attributes and modifiers) in a non-record type declaration, and treat it like we would the code of an instance field initializer?
+
+#### Properties
 
-If this `public` property is not explicitly declared within the *class-body*, then the compiler produces the property as follows:
+For each record parameter of a record type declaration there is a corresponding `public` property member whose name and type are taken from the value parameter declaration. Its name is the *identifier* of the *record_property_name*, if present, or the *identifier* of the *record_parameter* otherwise. If no concrete (i.e. non-abstract) public property with a `get` accessor and with this name and type is explicitly declared or inherited, it is produced by the compiler as follows:
 
 * For a *record struct* or a `sealed` *record class*:
  * A `private` `readonly` field is produced as a backing field for a `readonly` property. Its value is initialized during construction with the value of the corresponding primary constructor parameter.
  * The property's `get` accessor is implemented to return the value of the backing field.
-* For an `abstract` *record class*:
- * The property is declared `abstract` with only a `get` accessor.
+ * Each "matching" inherited virtual property's `get` accessor is overridden.
```

Comment:
In other words, in `class Point(int X, int Y): Base, Interface`, if there is a `virtual` property `X` of type `int` in either `Base` or `Interface`, they are implemented by `Point.X`.